iac-extensions/spring-config/src/main/java/org/sonar/iac/springconfig/parser/yaml/SpringConfigYamlPreprocessor.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * SonarQube IaC Plugin
+ * Copyright (C) 2021-2024 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.iac.springconfig.parser.yaml;
+
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+public class SpringConfigYamlPreprocessor {
+  private static final String DOCUMENT_BREAK = "---";
+  private static final int DOCUMENT_BREAK_LENGTH = DOCUMENT_BREAK.length();
+
+  public String preprocess(String source) {
+    return Stream.of(source.split("\n", -1))
```

Comment:
Can't we have other line breaks? I remember in Helm we had `List.of("\n", "\r\n", "\r", "\u2028", "\u2029");` to separate the files line by line